### ISSUE TITLE
Web terminal could not access in cluster resources

### ISSUE THREAD

**Issue by toschneck** [kubermatic/kubermatic#12899]
### What happened?
Currently when we create the web terminal container, a default network policy get applied, what drops in cluster traffic. So if I'm a developer and want quickly test my application for reachability via curl, I can't do this. In my opinion, the inside cluster traffic should be allowed. At least as a user I would expect it.
![Cluster 'hardcore-hodgkin-cilium-ingress-test-ympkwbs4k9' in Project 'Tobi's Demo Project 🚀' 2023-12-12 11-51-34](https://github.com/kubermatic/kubermatic/assets/4659261/458a79c2-8ccf-4bd6-965a-891551d8161a)
![Hubble UI 2023-12-12 11-52-21](https://github.com/kubermatic/kubermatic/assets/4659261/f0c0db26-f6a3-4e61-be52-5854b81842ab)

Current applied network policy:
`k get networkpolicies.networking.k8s.io  -n kube-system webterminal-7557a12007cf6f5291f9863fa086c056 -o yaml`
```yaml
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  creationTimestamp: "2023-12-12T09:50:46Z"
  generation: 1
  name: webterminal-7557a12007cf6f5291f9863fa086c056
  namespace: kube-system
  resourceVersion: "13770"
  uid: 5331f58b-3e82-4a79-ba9d-a6cc2aba5e18
spec:
  egress:
  - ports:
    - port: 30051
      protocol: TCP
    to:
    - ipBlock:
        cidr: 35.241.213.147/32
  - ports:
    - port: 443
      protocol: TCP
    to:
    - ipBlock:
        cidr: 10.241.0.1/32
  - ports:
    - port: 53
      protocol: TCP
    - port: 53
      protocol: UDP
    to:
    - ipBlock:
        cidr: 0.0.0.0/0
  podSelector:
    matchLabels:
      app: webterminal
  policyTypes:
  - Ingress
  - Egress
status: {}
```

### Expected behavior
Webterminal can access in cluster services / IPs (if no other network policy is applied)

### How to reproduce the issue?
Creat a cluster with and create a webservice in a namespace, e.g. the echo application at run-2
```
ingress:
  enabled: true
  hosts:
    - paths:
        - /test
  pathType: Prefix
```
Start Webterminal => Curl to endpoint what you get:
```bash
k get svc -n echoserver
curl 10.241.X.X
```

If you do the same with e.g. another debug container it works:
```bash
kubectl run shell-4033 --timeout 600s --namespace default --rm -i --tty --image nicolaka/netshoot -- /bin/sh -c bash
#commands from above
```

### How is your environment configured?

- KKP version: 2.24.0
- Shared or separate master/seed clusters?: combined

### Provide your KKP manifest here (if applicable)
see https://github.com/kubermatic/sig-cs-infra/tree/main/lab.kubermatic.io/run-2.lab.kubermatic.io/settings

### What cloud provider are you running on?

doesn't matter

### What operating system are you running in your user cluster?
Ubuntu

**Comment by embik** [comment]
> In my opinion, the inside cluster traffic should be allowed.

I disagree with this because it has security implications - the web terminal uses your OIDC identity and as such, you might just have access to specific namespaces via kubectl. If the web terminal is allowed traffic inside the cluster, users can access applications they don't have any permissions for.

We can maybe document how to create a NetworkPolicy that allows the web-terminal to access all (or specific) namespaces, but as a default this is potentially dangerous.

**Comment by toschneck** [comment]
Sounds also like a good idea. Couldn't we may have a setting on the webterminal "enable cluster network access" from the KKP UI?
And if not enabled some hint text that may some access is blocked.

**Comment by mfahlandt** [comment]
I think we should offer an option to at least allow egress traffic of the we terminal to make usage easier.
Otherwise you manually need to provide a network policy
E.g. 
` 
apiVersion: networking.k8s.io/v1
 kind: NetworkPolicy
 metadata:
   name: allow-webterminal-internet
   namespace: kube-system
 spec:
   podSelector:
     matchLabels:
       app: webterminal 
   policyTypes:
   - Egress
   egress:
   - {} # Allow all outbound traffic
`

**Comment by toschneck** [comment]
Agree, potential we could make a checkbox what enables the traffic
- [ ] allow external traffic for terminal

**Comment by csengerszabo** [comment]
It's a feature request at this point, not a bug

**Comment by toschneck** [comment]
It's implemented with `Enable WebTerminal Internet Access`
